src/Paprika/Merkle/RlpMemo.cs

```diff
@@ -64,139 +77,118 @@ public bool TryGetKeccak(byte nibble, out ReadOnlySpan<byte> keccak)
         return false;
     }
 
-    private Span<byte> GetAtNibble(byte nibble) => _buffer.Slice(nibble * Keccak.Size, Keccak.Size);
-
-    public static RlpMemo Decompress(scoped in ReadOnlySpan<byte> leftover, NibbleSet.Readonly children,
-        scoped in Span<byte> workingSet)
+    public bool Exists(byte nibble, NibbleSet.Readonly children)
     {
-        if (_decompressionForbidden)
-        {
-            ThrowDecompressionForbidden();
-        }
-
-        var span = workingSet[..Size];
-
-        if (leftover.IsEmpty)
-        {
-            // no RLP cached yet
-            span.Clear();
-            return new RlpMemo(span);
-        }
-
-        if (leftover.Length == Size)
+        // Check if the element exists
+        if (_buffer.Length == 0 || ((ushort)children & (1U << nibble)) == 0)
```

Comment:
Does it effectively check whether the child is set in children? If yes, isn't it `RlpMemo` agnostic and `NibbleSet.Readonly[nibble]` could be used directly in the call site? Then, have this method removed?